jruby-gradle-jar-plugin/src/main/groovy/com/github/jrubygradle/jar/JRubyJarTask.groovy

```diff
@@ -16,7 +22,196 @@ import org.gradle.api.tasks.StopExecutionException
 
+    enum Type { RUNNABLE, LIBRARY }
+
+    static final String DEFAULT_MAIN_CLASS = 'de.saumya.mojo.mains.JarMain'
+    static final String EXTRACTING_MAIN_CLASS = 'de.saumya.mojo.mains.ExtractingMain'
+
+    @Input
+    String jrubyVersion = project.jruby.defaultVersion
+    @Input
+    String jrubyMainsVersion = '0.2.0'
+    @Input
+    String mainClass
+    
+    /** Adds a GEM installation directory
+     */
+    @InputDirectory
+    void gemDir(def properties=[:],File f) {
+        gemDir(properties,f.absolutePath)
+    }
+
+    /** Adds a GEM installation directory
+     * @param Properties that affect how the GEM is packaged in the JAR. Currently only {@code fullGem} is
+     * supported. If set the full GEM content will be packed, otherwise only a subset will be packed.
+     * @param dir Source folder. Will be handled by {@code project.files(dir)}
+    */
+    void gemDir(def properties=[:],Object dir) {
+        CopySpec spec = GemUtils.gemCopySpec(properties,project,dir)
+        spec.exclude '.jrubydir'
+        with spec
+        // TODO change base plugin to store jars in
+        // project.jruby.gemInstallDir + '/jars' since they belong together
+        with GemUtils.jarCopySpec(project, {project.jruby.jarInstallDir})
+    }
+
+    /** Makes the JAR executable by setting a custom main class
+     *
+     * @param className Name of main class
+     */
+    @Incubating
+    void mainClass(final String className) {
+        this.mainClass = className
+        if (this.scriptName == null) {
+            this.scriptName = runnable()
+        }
+    }
+
+    @Incubating
+    void initScript(final Object scriptName) {
+        this.scriptName = scriptName
+    }
+
+    /** Sets the defaults
+     *
+     * @param defs A list of defaults. Currently {@code gems} and {@code mainClass} are the only recognised values.
+     * Unrecognised values are silently discarded
+     */
+    void defaults(final String... defs ) {
+        defs.each { String it ->
+            switch(it) {
+                case 'gems':
+                case 'mainClass':
+                    "default${it.capitalize()}"()
+            }
+        }
+    }
+
+    /** Loads the default GEM installation directory and
+     * JAR installation directory
+     *
+     */
+    void defaultGems() {
+        gemDir({project.jruby.gemInstallDir})
+    }
+
+    /** Makes the executable by adding a default main class
+     *
+     */
+    @Incubating
+    void defaultMainClass() {
+        mainClass(DEFAULT_MAIN_CLASS)
+    }
+
+    /** Makes the executable by adding a default main class
+     * which extracts the jar to temporary directory
+     *
+     */
+    @Incubating
+    void extractingMainClass() {
+        mainClass(EXTRACTING_MAIN_CLASS)
+    }
+
+    void jrubyVersion(String version) {
+      this.jrubyVersion = version
+    }
+
+    void jrubyMainsVersion(String version) {
+      this.jrubyMainsVersion = version
+    }
+
+    void jruby(Closure cfg) {
```

Comment:
Nice. Might add deprecation annotation as well as ading the `deprecated` word somewhere in the log message.